### PR TITLE
Feat/voicevox download file to local

### PR DESCRIPTION
**Pull request by Rob117**
Addresses #4 
This mostly consolidates code that I wrote in a separate file.

Get a VV audio query (first, mandatory API step)
Get a VV wav audio file using the results of the first query
Download the file to local under output.wav

### CODE REVIEW

File: app/main.go [hunk 1]
```diff
@@ -2,38 +2,58 @@ package main
...
 	"log"
 	"os"
 
 	"github.com/joho/godotenv"
 )
 
 // flags
-// if --Block, use block responses
-var isBlockFormat = flag.Bool("block", false, "Use non-streaming block responses")
+// if --VoiceVox, use block responses
```
Comment (risafj) [inline app/main.go]:
```suggestion
// if --vv, use block responses
```

File: app/main.go [hunk 2]
```diff
@@ -2,38 +2,58 @@ package main
...
 	openaiAPIKey := os.Getenv(openAiApiKey)
+	voiceVoxApiUrl := os.Getenv("VOICEVOX_API_URL")
 	if openaiAPIKey == "" {
 		log.Fatalf("%s not set", openAiApiKey)
 	}
 	flag.Parse()
-	client := CreateChatClient(openaiAPIKey, *isBlockFormat)
-	if *isBlockFormat {
+	chatClient := CreateChatClient(openaiAPIKey, *isVoiceVox)
+	if *isVoiceVox {
```
Comment (risafj) [inline app/main.go:29]: Let's use a guard clause here.
Or actually, you suggested abstracting this code away from main.go.

File: app/main.go [hunk 3]
```diff
@@ -2,38 +2,58 @@ package main
...
+				log.Fatalf("Error creating the output file: %v", err)
+				return
+			}
+			// Copy the response body to the output file
+			_, err = io.Copy(outputFile, audioClip)
+			if err != nil {
+				log.Fatalf("Error copying the response body to the output file: %f", err)
+				return
+			}
+			outputFile.Close()
```
Comment (risafj) [inline app/main.go]: Handle file closing error

File: app/voicevox_client.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	"net/http"
+	"net/url"
+)
+
+type voiceVoxClient struct {
+	baseUrl string
+	cid     string
+}
+
+func CreateVoiceVoxClient(baseUrl, cid string) *voiceVoxClient {
```
Comment (risafj) [inline app/voicevox_client.go]: Let's export the return type

File: app/main.go [hunk 5]
```diff
@@ -2,38 +2,58 @@ package main
...
 	loadEnv()
 	openaiAPIKey := os.Getenv(openAiApiKey)
+	voiceVoxApiUrl := os.Getenv("VOICEVOX_API_URL")
 	if openaiAPIKey == "" {
 		log.Fatalf("%s not set", openAiApiKey)
 	}
 	flag.Parse()
-	client := CreateChatClient(openaiAPIKey, *isBlockFormat)
-	if *isBlockFormat {
+	chatClient := CreateChatClient(openaiAPIKey, *isVoiceVox)
```
Comment (risafj) [inline app/main.go]: Let's export the return type of CreateChatClient

File: app/voicevox_client.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+func (c *voiceVoxClient) getAudioQuery(text string) (io.ReadCloser, error) {
+	// Make an http request to BASE_URL + AUDIO_QUERY_ENDPOINT
+	params := url.Values{}
+	params.Add("speaker", c.cid)
+	params.Add("text", text)
+	u, _ := url.ParseRequestURI(c.baseUrl)
+	u.Path = "/audio_query"
+	u.RawQuery = params.Encode()
+	urlStr := fmt.Sprintf("%v", u)
+	res, e := http.Post(urlStr, "application/json", nil)
```
Comment (risafj) [inline app/voicevox_client.go]: Is it possible to slim down the boilerplate code?

Comment (Rob117) [reply app/voicevox_client.go]: This ammendment was made with papa chats

**Review by risafj — APPROVED** [review]

**Pull request merged**